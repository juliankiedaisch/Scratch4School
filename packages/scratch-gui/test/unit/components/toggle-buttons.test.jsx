import React from 'react';
import ToggleButtons from '../../../src/components/toggle-buttons/toggle-buttons';
import {fireEvent, render} from '@testing-library/react';


describe('ToggleButtons', () => {
    const buttons = [
        {
            title: 'Button 1',
            handleClick: jest.fn(),
            icon: 'Button 1 icon'
        },
        {
            title: 'Button 2',
            handleClick: jest.fn(),
            icon: 'Button 2 icon'
        }
    ];

    test('renders multiple buttons', () => {
        const {container} = render(<ToggleButtons
            buttons={buttons}
        />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('calls correct click handler', () => {
        const {container} = render(<ToggleButtons
            buttons={buttons}
        />);

        const button2 = container.querySelector('button[title="Button 2"]');
        fireEvent.click(button2);
        
        expect(buttons[1].handleClick).toHaveBeenCalled();
        expect(buttons[0].handleClick).not.toHaveBeenCalled();
    });
});
