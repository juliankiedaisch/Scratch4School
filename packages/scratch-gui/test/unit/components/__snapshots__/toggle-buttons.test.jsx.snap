// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButtons renders multiple buttons 1`] = `
<div
  class=""
>
  <button
    aria-label="Button 1"
    title="Button 1"
  >
    <img
      aria-hidden="true"
      src="Button 1 icon"
    />
  </button>
  <button
    aria-label="Button 2"
    title="Button 2"
  >
    <img
      aria-hidden="true"
      src="Button 2 icon"
    />
  </button>
</div>
`;
